Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -154,15 +154,15 @@ void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DO
         }
 
         if (!isAllowedByFeaturePolicy(*document, permissionDescriptor.name)) {
-            promise.resolve(PermissionStatus::create(*context, PermissionState::Denied, permissionDescriptor));
+            promise.resolve(PermissionStatus::create(*context, PermissionState::Denied, permissionDescriptor, PermissionQuerySource::Window, document->page()));
             return;
         }
 
         PermissionController::shared().query(ClientOrigin { document->topOrigin().data(), WTFMove(originData) }, permissionDescriptor, document->page(), *source, [document = Ref { *document }, permissionDescriptor, promise = WTFMove(promise)](auto permissionState) mutable {
             if (!permissionState)
                 promise.reject(Exception { NotSupportedError, "Permissions::query does not support this API"_s });
             else
-                promise.resolve(PermissionStatus::create(document, *permissionState, permissionDescriptor));
+                promise.resolve(PermissionStatus::create(document, *permissionState, permissionDescriptor, PermissionQuerySource::Window, WeakPtr { document->page() }));
```

Comment:
`WeakPtr { }` is likely not needed here since you omitted it a few lines above.
Also, `*document->page()` since we know it is non-null and we're constructing a WeakPtr.